### PR TITLE
Add support for deleting and deprecating pods

### LINKED ISSUE #31: Provide a subcommand to yank/deprecate a pod
https://github.com/CocoaPods/CocoaPods/issues/2235

We would need changes to trunk API to provide support for this.

### PR DESCRIPTION
**Pull request by segiddins**
Closes #31.

### CODE REVIEW

File: lib/pod/command/trunk/delete.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        end
+
+        def validate!
+          super
+          help! 'Please specify a pod name.' unless @name
+          help! 'Please specify a version.' unless @version
+        end
+
+        def run
+          json = delete
```
Comment (segiddins) [inline lib/pod/command/trunk/delete.rb]: should this ask for confirmation?

Comment (kylef) [inline lib/pod/command/trunk/delete.rb]: I think that might be a good idea, and also a chance to show warning again:

> WARNING: It is generally considered bad behavior to remove versions of a Pod that others are depending on! Please consider using the deprecate command instead.

**Comment by segiddins** [comment]
@orta please give this a quick review?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,16 @@
 ## Master
 
+##### Enhancements
+
+* The `pod deprecate PODNAME` command has been added to deprecate all versions
+  of a pod.
+  [Samuel Giddins](https://github.com/segiddins)
+  [#31](https://github.com/CocoaPods/cocoapods-trunk/issues/31)
+
+* The `pod deprecate PODNAME VERSION` command has been added to delete a single
```
Comment (orta) [inline CHANGELOG.md]: the comment here is "delete", is this the right terminology?

File: lib/pod/command/trunk/delete.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        def run
+          return unless confirm_deletion?
+          json = delete
+          print_messages(json['data_url'], json['messages'])
+        end
+
+        private
+
+        def confirm_deletion?
+          warning_message = 'WARNING: It is generally considered bad behavior ' \
```
Comment (orta) [inline lib/pod/command/trunk/delete.rb]: if you wanted to code-gold, this message could be a static

File: lib/pod/command/trunk/delete.rb [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+            'Please consider using the `deprecate` command instead.'
+          UI.puts(warning_message.yellow)
+          loop do
+            UI.print("Are you sure you want to delete this Pod version?\n> ")
+            answer = UI.gets.strip.downcase
+            UI.puts # ensures a newline is printed after the user input
+            affirmatives = %w(y yes true 1)
+            negatives = %w(n no false 0)
+            return true if affirmatives.include?(answer)
+            return false if negatives.include?(answer)
```
Comment (orta) [inline lib/pod/command/trunk/delete.rb]: I like this format for input

**Comment by orta** [comment]
only the CHANGELOG thing is a semi-blocker :+1:

**Comment by segiddins** [comment]
Sweeeeet. :shipit:

**Pull request merged**